### PR TITLE
Add support for deb packages

### PR DESCRIPTION
**Pull request by tiagonux**
Hi,

These commit series add support for deb packages, the main idea is to allow the users to use its own custom OVN/OVS deb packages.

### CODE REVIEW

File: image/ovn/Dockerfile [hunk 1]
```diff
@@ -20,11 +22,19 @@ COPY run_ovsdb_etcd.sh /run_ovsdb_etcd.sh
 COPY run_ovsdb_etcd_sb.sh /run_ovsdb_etcd_sb.sh
 COPY run_ovsdb_etcd_nb.sh /run_ovsdb_etcd_nb.sh
 
-RUN /install_ovn.sh $USE_OVN_RPMS $EXTRA_OPTIMIZE
+RUN /install_ovn.sh $USE_OVN_RPMS $USE_OVN_DEBS $EXTRA_OPTIMIZE
 RUN /install_utils_from_sources.sh $INSTALL_UTILS_FROM_SOURCES
 RUN /install_ovsdb_etcd.sh
 RUN /install_etcd.sh
 
+RUN if [ "$USE_OVN_DEBS" = "yes" ]; then \
```
Comment (dceara) [inline image/ovn/Dockerfile:30]: Why do we need to do this only when installing deb packages?  Doesn't this make sense for both rpm-based and deb-based distros?

Comment (tiagonux) [reply image/ovn/Dockerfile]: Hey @dceara makes sense. I did the fix.

**Comment by tiagonux** [comment]
I just added a logic to cover it.

File: image/ovn/Dockerfile (hunk 1 above)
Comment (tiagonux) [reply image/ovn/Dockerfile:30]: Sorry, I broke something I will fix in the next commit.

Comment (tiagonux) [reply image/ovn/Dockerfile]: hey @dceara could you follow with the review? Thanks

**Comment by tiagonux** [comment]
Hey @dceara could you review it?

Thanks

Tiago Pires

**Comment by numansiddique** [comment]
@tiagonux  Can you please add your Signed-off-by tag to the commit message in this PR ?

Please see "git log" to see an example if you're not aware of this tag usage.

Thanks

**Comment by tiagonux** [comment]
> @tiagonux Can you please add your Signed-off-by tag to the commit message in this PR ?
> 
> Please see "git log" to see an example if you're not aware of this tag usage.
> 
> Thanks

Hi @numansiddique, it is done.

Thanks

Tiago Pires

**Review by dceara — CHANGES_REQUESTED** [review]
Sorry for the delay in reviewing this.

Overall it looks OK but I have a few minor comments/questions.

Thanks,
Dumitru

File: image/cinc/install_fedora_pkg.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ systemctl mask \
 	systemd-udevd.service \
 	systemd-vconsole-setup.service
 
-dnf -y --skip-broken install \
+dnf -y install --skip-broken  \
```
[the remark this answers is not in the document]
Comment (tiagonux) [reply image/cinc/install_fedora_pkg.sh]: It is done. Tks

File: ovn_cluster.sh [hunk 3]
```diff
@@ -964,7 +965,7 @@ function build-images() {
 
     ${RUNC_CMD} build -t ovn/ovn-multi-node --build-arg OVS_SRC_PATH=ovs \
     --build-arg OVN_SRC_PATH=ovn --build-arg USE_OVN_RPMS=${USE_OVN_RPMS} \
-    --build-arg EXTRA_OPTIMIZE=${EXTRA_OPTIMIZE} \
+    --build-arg USE_OVN_DEBS=${USE_OVN_DEBS} --build-arg EXTRA_OPTIMIZE=${EXTRA_OPTIMIZE} \
```
[the remark this answers is not in the document]
Comment (tiagonux) [reply ovn_cluster.sh]: It is done. Tks

File: ovn_cluster.sh [hunk 4]
```diff
@@ -1114,10 +1129,25 @@ case "${1:-""}" in
         ;;
     build)
         check-for-ovn-rpms
+        check-for-ovn-debs
         if [ "$USE_OVN_RPMS" == "yes" ]
         then
+            if [ "$USE_OVN_DEBS" == "yes" ]
+            then
+                echo "Do not keep rpm and deb packages on the same directory or enable both package manager!"
```
[the remark this answers is not in the document]
Comment (tiagonux) [reply ovn_cluster.sh]: It is done. Tks

File: image/ovn/Dockerfile [hunk 5]
```diff
@@ -20,11 +22,19 @@ COPY run_ovsdb_etcd.sh /run_ovsdb_etcd.sh
 COPY run_ovsdb_etcd_sb.sh /run_ovsdb_etcd_sb.sh
 COPY run_ovsdb_etcd_nb.sh /run_ovsdb_etcd_nb.sh
 
-RUN /install_ovn.sh $USE_OVN_RPMS $EXTRA_OPTIMIZE
+RUN /install_ovn.sh $USE_OVN_RPMS $USE_OVN_DEBS $EXTRA_OPTIMIZE
 RUN /install_utils_from_sources.sh $INSTALL_UTILS_FROM_SOURCES
 RUN /install_ovsdb_etcd.sh
 RUN /install_etcd.sh
 
+RUN if [ "$USE_OVN_RPMS" = "yes" ] || [ "$USE_OVN_DEBS" = "yes" ] ; then \
```
[the remark this answers is not in the document]
Comment (tiagonux) [reply image/ovn/Dockerfile:33]: It is done. Tks

**Comment by tiagonux** [comment]
Hi @dceara 

The fixes were done.

Thanks

Tiago Pires

**Review by dceara — APPROVED** [review]
Looks good to me, thanks!

**Comment by numansiddique** [comment]
@tiagonux Thanks for the contribution

**Pull request merged**